### DISCUSSION TITLE
When my input and selection are in the child component, how can my parent component get the value?

### DISCUSSION THREAD

**Discussion by zhushenhang** [react-hook-form/react-hook-form#3870] [Q&A] [answered]
Hello,my English is not good.please excuse me.
for example:
TaskForm.js is my parent component
```
const TaskForm = (props) => {
  const { handleSubmit, errors, control } = useForm();
  const classes = useStyles();
  const { onSubmit } = props;

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <ContentField
        legendClassName={classes.legend}
        fieldSetClassName={classes.fieldSet}
        title='input'
        anchorId='input'
      >
        <TaskInput />
      </ContentField>
      <Button variant='contained' color='primary' type='submit'>
        submit
      </Button>
    </form>
  );
};

export default TaskForm;
```
TaskInput.js is my child
```
const TaskInput = (props) => {
  const classes = useStyles();
  const [selectedValue, setSelectedValue] = useState('udp');

  const setRadio = (event) => {
    setSelectedValue(event.target.value);
  };

  return (
    <div className={classes.root}>
      <InputType selectedValue={selectedValue} setRadio={setRadio} />
      <InputUdp />
      {/* {getParam(selectedValue)} */}
    </div>
  );
};
```
InputUdp.js
```
const InputUdp = (props) => {
  const { control } = useForm();
  return (
    <Grid container>
      <Grid item xs={12} sm={6} lg={3} xl={3}>
        <Controller
          as={TextField}
          name='localIp'
          label='IP'
          defaultValue=''
          control={control}
        />
      </Grid>
    </Grid>
  );
};
```
When I click Submit, I cannot get the value of InputUdp.what should I do?
thanks

**Comment by bluebill1049** [comment, marked as answer]
you should `useFormContext` to get the Contro, not initialize another `useForm`, you are creating two forms in your example, or simply pass control down as prop.

**Reply by zhushenhang** [reply]
thanks bro. Pass control down as prop can work for me. 
But when I use checked, the value is undefined. I need to annotate checked and manually click radio to get the value. Why?
Or do I need to use another API?
```
<Controller
        as={
          <RadioGroup name='inputTypeRadio' row>
            <FormControlLabel
              value='udp'
              control={<Radio size='small' required />}
              label='UDP'
              // checked
            />
          </RadioGroup>
        }
        name='inputTypeRadio'
        control={control}
        defaultValue=''
      />
```

**Reply by zhushenhang** [reply]
Oh, I see. Now it's controlled by defaultValue

**Reply by zhushenhang** [reply]
Thank you again for your help

**Reply by bluebill1049** [reply]
no worries mate.

**Reply by Priya-dev** [reply]
@[zhushenhang](https://github.com/zhushenhang) Can you please share the codesandbox if you have any? I have the same issue.

**Reply by zhushenhang** [reply]
Sorry bro.
I have no codesandbox . 
You can tell us your problem and  try our best to help you

**Reply by Priya-dev** [reply]
@zhushenhang @bluebill1049 
I have created a code sandbox here
https://codesandbox.io/s/react-hook-form-useformcontext-forked-ver9h2?file=/src/App.js:226-254

The parent component has a form with a couple of inputs and the child component has another form with a lot of input fields.
Note that the child component is re-used somewhere else and hence I have to have one form in the parent and another in the child.
The issue is the submit button in the parent field is enabled even though the fields in the child component are required fields. Until the input in the child component field is clicked, the button stays enabled.
